### DISCUSSION TITLE
GitHub work week update

### DISCUSSION THREAD

**Discussion by pransh15** [mdn/mdn-community#708] [Announcements]
Hi Community 👋

The MDN team held a GitHub work week in the week of July 8th. We are very happy with what we achieved with what we had planned.

### TL;DR
- We held a GitHub work week to focus on how we work in sync with the community on GitHub.
- Focused on Stale PRs and Issues across the org.
- Discussed issue triage guidelines.
- Reworked teams to a new system that organizes them via team (e.g., MDN, Contributors, Peers, etc.). That includes removal of some teams like `yari-content-{locale}`, `mdn-{function}`, etc.
- Reviewed access rights to all repositories.
- Evaluated our current pull request review process, received feedback from community members, and explored options for improvement.
- Closed/responded to old discussions.

We divided the work into different days and concentrated on an area each day:

### Monday - Stale PRs & Issues
Reviewed PRs and Issues across the organization to find resolutions for anything before 2023. Closed stale or invalid items, encouraged relevant ones to continue the conversation or create new PRs/Issues. Developed a recommended issue triaging process:

- Validate issues, then label for the community (good-first-issue, help-wanted, accepting PRs).
- If not a good first issue, assign to the content team or mark for discussion.
- Apply appropriate labels for priority and effort.
- Leave a comment with the next steps.
- If stuck, create a new discussion.

We will add detailed guidelines to the docs this week.

### Tuesday - Teams & Roles
Reorganized teams for easier management and understanding. New structure:
-  MDN
-- Community, Content, Engineering, Product, QA
-  Contributors
-- Reviewers, Members    
-  Partners
-- Experts, Contractors, Survey Partners    
-  Localization
-- Team Leads
-- DE, ES, FR, JA, KO, BR, RU, ZH    
-  Peers
-- SRE, Webdriver, Web Extensions, BCD
-  Bots

Feel free to let us know if you have any questions or see a needed change in your team.

### Wednesday - Pull Request Review Process
Identified problems and solutions:

- Too many people being assigned: Simplify the codeowners file.
- Unassigned reviews: Comments are fine, but only use the review mechanism to request changes.
- One reviewer per PR: To ensure a smooth and focused review process, please assign only one reviewer per PR. Having multiple reviewers can lead to disorganized conversations and occasionally expands the scope.
- Template issues: Motivation and Scope fields are now required.
- Time constraints: Include timeline in the scope, and use priority labels.
- Auto-assign PRs: We will remove you from a round-robin team if needed, and add you to specific individual category-based assignments. Just reply to this discussion, and we can discuss this.
- Firefox release notes reviews: Core team will address these PRs.
- Reviewer guidelines: Link guidelines in the PR template and GitHub Action.

### Thursday - Discussion Triage Process
- Triaged discussions, closed outdated or resolved ones from before 2023. Valid discussions are still open and awaiting responses.
- Will update community guidelines on discussions and consider removing some categories.

### Friday - Issue Triage Process
Identified inconsistencies in issue triaging and created new guidelines for community members. Reviewed and wrapped up stale issues in repos other than content and yari.

Excited to announce a new project board for contributors who want to help triage issues and improve MDN together! 🎉 

**Thank you for being with us and making MDN a trusted place for learning about building on the web!** ♥️
